src/components/CallView/shared/EmptyCallView.vue

```diff
@@ -148,19 +160,36 @@ export default {
 		},
 
 		helper() {
-			if (this.connectionFailed) {
-				return t('spreed', 'Please try to reload the page')
-			}
-			return ''
+			return this.connectionFailed ? t('spreed', 'Please try to reload the page') : ''
 		},
 
 		message() {
-			if (this.connectionFailed === 'consent') {
-				return t('spreed', 'Recording consent is required')
-			}
-
 			if (this.connectionFailed) {
-				return t('spreed', 'Something went wrong')
+				let errorReason = ''
+				switch (this.connectionFailed.meta?.statuscode) {
+				case 0:
+					errorReason = t('spreed', 'No response from server')
+					break
+				case 400:
+					errorReason = t('spreed', 'Recording consent is required')
+					break
+				case 403:
+					errorReason = t('spreed', 'This conversation is read-only')
+					break
+				case 404:
+					errorReason = t('spreed', 'Conversation not found or not joined')
+					break
+				case 412:
+					errorReason = t('spreed', "Lobby is still active and you're not a moderator")
+					break
+				default:
+					if (this.connectionFailed.data?.error) {
+						errorReason = this.connectionFailed.data.error
+					} else {
+						errorReason = t('spreed', 'Unknown error occured')
+					}
+				}
+				return errorReason
```

Comment:
To get rid of switch:

```suggestion
				const statusErrors = {
					0: t('spreed', 'No response from server'),
					400: t('spreed', 'Recording consent is required'),
					403: t('spreed', 'This conversation is read-only'),
					404: t('spreed', 'Conversation not found or not joined'),
					412: t('spreed', "Lobby is still active and you're not a moderator"),
				}
				const statusCode = this.connectionFailed.meta?.statuscode
				return statusCode 
					? statusErrors[statusCode]
					: this.connectionFailed.data?.error
						|| t('spreed', 'Unknown error occured')
```

or

```suggestion
				const statusErrors = {
					0: t('spreed', 'No response from server'),
					400: t('spreed', 'Recording consent is required'),
					403: t('spreed', 'This conversation is read-only'),
					404: t('spreed', 'Conversation not found or not joined'),
					412: t('spreed', "Lobby is still active and you're not a moderator"),
				}

				const statusCode = this.connectionFailed.meta?.statuscode
				if (statusErrors[statusCode]) {
					return statusErrors[statusCode]
				}

				if (this.connectionFailed.data?.error) {
					return this.connectionFailed.data.error
				}
				
				return t('spreed', 'Unknown error occured')
```

`switch` works as a many `goto` wwith labels, being huge, with a requirement of `break` (that can be lost) and less declarative.

Reply:
BTW, can status code be 0 here?

Reply:
iOS is taking it into account, haven't checked if there is indeed such a response
https://github.com/nextcloud/talk-ios/blob/81ecc92ada12ae88f9a2f2aa6c297fa5295111d8/NextcloudTalk/NCCallController.m#L230-L257

Reply:
> iOS is taking it into account, haven't checked if there is indeed such a response

Status code 0 is an internal code from iOS, it is not a valid HTTP status code. On iOS it occurs on various reasons, for example timeout.

Reply:
Then I'd get rid of it, as we have `Unkown error occured` (Wasn't it `Something went wrong?`)